### DISCUSSION TITLE
Error 13801 while connecting several remote Windows 10 clients to Strongswan with lets-encrypt certificates (IKEv2-EAP)

### DISCUSSION THREAD

**Discussion by RusanovskyAlexey** [strongswan/strongswan#473] [Q&A]
Hello

I set up a Strongswan server for VPN clients to access the internal network (EAP-IKEv2). I have successfully configured it using Letsencrypt server certificates and it works for clients using Mac OS X, IOS, Windows 7 and Windows 10.

Everything was working fine for a year

But a few weeks ago several remote clients using windows 10 started getting error during the connection

Server: Strongswan version 5.8.2 on FreeBSD 11.2-RELEASE-p15 Client: Mac OS X (several versions) / IOS (several versions) / Windows 7 (several versions) / Windows 10 (several versions)

Windows 10 VPN error: 13801: IKE authentication credentials are unacceptable error

At the same time, other remote clients, including those using Windows 10 with the same build number, work fine.

The saddest thing is that the error does not correlate with the build number of Windows 10

Of course the certificate is extended and valid

You can find all the details below.

Thank you for your time. I would be grateful for any help

### ipsec.conf

```
config setup

  strictcrlpolicy=no
  charondebug="ike 1, knl 1, cfg 0"
  uniqueids=no


conn ikev2-vpn
  auto=add
  compress=no
  type=transport
  keyexchange=ikev2
  fragmentation=yes
  forceencaps=yes

  ike=aes128-sha256-ecp256,aes256-sha384-ecp384,aes128-sha256-modp2048,aes128-sha1-modp2048,aes256-sha384-modp4096,aes256-sha256-modp4096,aes256-sha1-modp4096,aes128-sha256-modp1536,aes128-sha1-modp1536,aes256-sha384-modp2048,aes256-sha2
  esp=aes128gcm16-ecp256,aes256gcm16-ecp384,aes128-sha256-ecp256,aes256-sha384-ecp384,aes128-sha256-modp2048,aes128-sha1-modp2048,aes256-sha384-modp4096,aes256-sha256-modp4096,aes256-sha1-modp4096,aes128-sha256-modp1536,aes128-sha1-modp1

  dpdaction=clear
  dpddelay=300s
  rekey=no

  left=%any
  leftid=@vpn.host.com
  leftcert=fullchain.pem
  leftsendcert=always
  leftsubnet=0.0.0.0/0

  right=%any
  rightid=%any
  rightauth=eap-mschapv2
  rightsourceip=192.168.20.2-192.168.20.50
  rightdns=192.168.70.253,192.168.70.254

  eap_identity=%identity
```

### final part of charon.log

```
Jun 23 09:12:17 11[MGR] <ikev2-vpn|10> checkin IKE_SA ikev2-vpn[10]
Jun 23 09:12:17 03[NET] sending packet: from *serverip*[4500] to *clientip*[4500]
Jun 23 09:12:17 11[MGR] <ikev2-vpn|10> checkin of IKE_SA successful
Jun 23 09:12:17 03[NET] sending packet: from *serverip*[4500] to *clientip*[4500]
Jun 23 09:12:46 06[NET] waiting for data on sockets
Jun 23 09:12:46 01[JOB] got event, queuing job for execution
Jun 23 09:12:46 01[JOB] next event in 628ms, waiting
Jun 23 09:12:46 11[MGR] checkout IKEv2 SA with SPIs 688d0386698d3362_i b3e60629dc447607_r
Jun 23 09:12:46 11[MGR] IKE_SA checkout not successful
Jun 23 09:12:47 01[JOB] got event, queuing job for execution
Jun 23 09:12:47 01[JOB] next event in 98s 38ms, waiting
Jun 23 09:12:47 11[MGR] checkout IKEv2 SA with SPIs ef71603dd0f2ce38_i 6e86dbaeb491d377_r
Jun 23 09:12:47 11[MGR] IKE_SA ikev2-vpn[10] successfully checked out
Jun 23 09:12:47 11[JOB] <ikev2-vpn|10> deleting half open IKE_SA with *clientip* after timeout
Jun 23 09:12:47 11[MGR] <ikev2-vpn|10> checkin and destroy IKE_SA ikev2-vpn[10]
Jun 23 09:12:47 11[IKE] <ikev2-vpn|10> IKE_SA ikev2-vpn[10] state change: CONNECTING => DESTROYING
Jun 23 09:12:47 11[MGR] checkin and destroy of IKE_SA successful
```

**Comment by tobiasbrunner** [comment]
Please do not [cross-post](https://serverfault.com/q/1067505/95913).

This could be related to [changes in the certificate chain](https://letsencrypt.org/2020/09/17/new-root-and-intermediates.html) of Let's Encrypt. The Windows clients where it doesn't work might not have the new root CA certificate yet if your chain uses that (they are loaded lazily from Microsoft's online trust store), or they don't have the new intermediate CA certificate in case you don't send it to your VPN clients (possible that Microsoft loads those too if necessary). If it's because of the intermediate CA certificate, make sure to send it to your clients (install it in `ipsec.d/cacerts`), or try browsing to a website protected by a Let's Encrypt certificate via Edge or IE so the missing CA certificates are loaded.

**Reply by RusanovskyAlexey** [reply]
Unfortunately, I had already considered this to be the cause of the problem.
Furthermore, I had installed the intermediate and root certificates in the trusted certificate store on the remote client machines from the fullchain.pem file.

P.S. I tried the option of opening a Letsencrypt-protected site in IE after your message.  Unfortunately, the problem is still there.
P.P.S. Also (I just checked) the intermediate and root certificates are in the chain.pem file in the cacerts directory.

**Reply by Thermi** [reply]
Hi, please verify if all the certificates in use are still valid (not expired, revoked, ...).

EDIT: Also, make sure that the leaf certificate is the first on in fullchain.pem. Otherwise it's not loaded.

**Reply by stampycode** [reply]
Hi, I'm having the same problem - been using the same config for over a year, multiple certificate renewals, but the latest round of clean-install Windows machines seem to be unable to successfully establish a connection with the vpn server, when other servers (and all other types of devices) can connect fine.
I'm suspecting it's as you say, a certificate problem on Windows side, but so far not been able to crack this.
Have manually installed a pile of Root and Intermediate certs from LetsEncrypt website for their new chain, but no success.
No errors reported in IPSec logs on server side, just connection timeouts from clients who display this error.

**Reply by stampycode** [reply]
Ok, got it - Needed to download the "ISRG Root X1" cert from the Lets Encrypt certificate page (linked above), and install it into the `Local Machine` certificate registry, in the store `Third Party Root Certification Authorities`.

I tried several variations of this, and the "automatically select the certificate store.." option does NOT work, and the certificate needs to be in the "Local Machine" and NOT the "Current User" certificate store.

I hope Microsoft pulls their finger out and fixes this. It's *extremely* unhelpful.

```
    Write-Host " Fetching Lets Encrypt root CA Cert..." -ForegroundColor Cyan
    $output = "isrgrootx1.der"
    $configUrl = "https://letsencrypt.org/certs/isrgrootx1.der"
    Invoke-WebRequest -Uri $configUrl -OutFile $output
    Import-Certificate -FilePath "$output" -CertStoreLocation 'Cert:\LocalMachine\Root'-Verbose
```

**Reply by madcyc** [reply]
Hi,
after visiting website with cert from letsencrypt with Ms Edge ISRG Root X1 has been stored into certlm.msc

But still I receive same error from Windows 10:
"error: 13801: IKE authentication credentials are unacceptable error"

In charon.log I can only find the following lines about intermediate cert R3:
Aug 12 10:50:17 11[IKE] <1> sending cert request for "C=US, O=Let's Encrypt, CN=R3"
Aug 12 10:50:17 11[IKE] <1> sending cert request for "C=US, O=Internet Security Research Group, CN=ISRG Root X1"

When I capture traffic with wireshark I can see strongSWAN sending fullchain.pem to Windows 10 client, but not R3.pem although it has been loaded without throwing an error in charon.log.
When I put R3 into certlm.msc IKE_SA will be established but I want Windows to load it automatically.
So how I can I do that?

**Reply by tobiasbrunner** [reply]
@madcyc Please post the complete log and your config (see #196).

**Reply by madcyc** [reply]
> ### /etc/ipsec.conf
> conn win10_letsencrypt
>     ike = aes256gcm16-sha2_384-prfsha384-modp1024,aes256-sha256-prfsha256-modp1024,aes256-sha512-prfsha512-modp1024
>     type=transport
>     keyexchange=ikev2
>     left=%defaultroute
>     left=%any
>     leftcert=fullchain.pem
>     leftsendcert=always
>     leftsubnet=0.0.0.0/0
>     leftauth=pubkey
>     leftid=@vpn4rpi.hopto.org
>     rightsourceip=10.10.3.0/24
>     rightauth=eap-mschapv2
>     right=%any
>     eap_identity=%any
>     auto=add

[charon_debug.log](https://github.com/strongswan/strongswan/files/6976577/charon_debug.log)

**Reply by tobiasbrunner** [reply]
Could you please add the output of `ipsec listcerts` and `ipsec listcacerts`.

**Reply by madcyc** [reply]
# ipsec listcerts

> List of X.509 End Entity Certificates
> 
>   subject:  "CN=vpn4rpi.hopto.org"
>   issuer:   "C=US, O=Let's Encrypt, CN=R3"
>   validity:  not before Jul 27 10:39:27 2021, ok
>              not after  Oct 25 10:39:25 2021, ok (expires in 73 days)
>   serial:    04:79:7b:b4:96:eb:b4:69:e3:cb:e8:8c:b4:ba:c9:47:d7:ec
>   altNames:  vpn4rpi.hopto.org
>   flags:     serverAuth clientAuth
>   OCSP URIs: http://r3.o.lencr.org
>   certificatePolicies:
>              2.23.140.1.2.1
>              1.3.6.1.4.1.44947.1.1.1
>              CPS: http://cps.letsencrypt.org
>   authkeyId: 14:2e:b3:17:b7:58:56:cb:ae:50:09:40:e6:1f:af:9d:8b:14:c2:c6
>   subjkeyId: 8b:38:7b:f8:6f:38:8d:6a:33:07:80:38:56:90:74:14:70:5f:f0:0b
>   pubkey:    RSA 2048 bits, has private key
>   keyid:     8e:83:77:ce:4e:8a:62:94:fe:ca:a2:dc:50:be:44:58:29:26:a1:3e
>   subjkey:   8b:38:7b:f8:6f:38:8d:6a:33:07:80:38:56:90:74:14:70:5f:f0:0b
> 

# ipsec listcacerts

> List of X.509 CA Certificates
> 
>   subject:  "C=US, O=Let's Encrypt, CN=R3"
>   issuer:   "O=Digital Signature Trust Co., CN=DST Root CA X3"
>   validity:  not before Oct 07 20:21:40 2020, ok
>              not after  Sep 29 20:21:40 2021, ok (expires in 48 days)
>   serial:    40:01:75:04:83:14:a4:c8:21:8c:84:a9:0c:16:cd:df
>   flags:     CA CRLSign serverAuth clientAuth
>   CRL URIs:  http://crl.identrust.com/DSTROOTCAX3CRL.crl
>   pathlen:   0
>   certificatePolicies:
>              2.23.140.1.2.1
>              1.3.6.1.4.1.44947.1.1.1
>              CPS: http://cps.root-x1.letsencrypt.org
>   authkeyId: c4:a7:b1:a4:7b:2c:71:fa:db:e1:4b:90:75:ff:c4:15:60:85:89:10
>   subjkeyId: 14:2e:b3:17:b7:58:56:cb:ae:50:09:40:e6:1f:af:9d:8b:14:c2:c6
>   pubkey:    RSA 2048 bits
>   keyid:     8a:93:82:f4:c8:04:08:34:5e:5b:c2:f8:d7:55:d3:c2:e7:62:48:cf
>   subjkey:   14:2e:b3:17:b7:58:56:cb:ae:50:09:40:e6:1f:af:9d:8b:14:c2:c6
> 
>   subject:  "C=US, O=Internet Security Research Group, CN=ISRG Root X1"
>   issuer:   "C=US, O=Internet Security Research Group, CN=ISRG Root X1"
>   validity:  not before Jun 04 12:04:38 2015, ok
>              not after  Jun 04 12:04:38 2035, ok (expires in 5043 days)
>   serial:    82:10:cf:b0:d2:40:e3:59:44:63:e0:bb:63:82:8b:00
>   flags:     CA CRLSign self-signed
>   subjkeyId: 79:b4:59:e6:7b:b6:e5:e4:01:73:80:08:88:c8:1a:58:f6:e9:9b:6e
>   pubkey:    RSA 4096 bits
>   keyid:     f8:16:51:3c:fd:1b:44:9f:2e:6b:28:a1:97:22:1f:b8:1f:51:4e:3c
>   subjkey:   79:b4:59:e6:7b:b6:e5:e4:01:73:80:08:88:c8:1a:58:f6:e9:9b:6e
> 
>   subject:  "C=CH, O=strongSwan, CN=strongswan.ca"
>   issuer:   "C=CH, O=strongSwan, CN=strongswan.ca"
>   validity:  not before Jul 21 22:52:04 2021, ok
>              not after  Jul 20 22:52:04 2024, ok (expires in 1073 days)
>   serial:    1c:12:a3:3a:c8:d9:81:f7
>   flags:     CA CRLSign self-signed
>   subjkeyId: c6:22:c4:d7:81:e2:c9:39:d2:f2:51:c8:3a:02:1f:9a:b7:78:2f:70
>   pubkey:    RSA 2048 bits
>   keyid:     b9:9e:00:6a:04:b4:b4:93:d3:e2:4e:07:c3:9f:11:80:59:ae:55:44
>   subjkey:   c6:22:c4:d7:81:e2:c9:39:d2:f2:51:c8:3a:02:1f:9a:b7:78:2f:70
> 
>   subject:  "C=CH, O=strongSwan, CN=strongSwan CA"
>   issuer:   "C=CH, O=strongSwan, CN=strongSwan CA"
>   validity:  not before Jul 14 15:53:34 2021, ok
>              not after  Jul 13 15:53:34 2024, ok (expires in 1065 days)
>   serial:    0f:3f:08:bf:42:39:99:dc
>   flags:     CA CRLSign self-signed
>   subjkeyId: 5a:42:1a:d4:e3:35:9a:c6:e2:4f:94:82:8b:7e:22:08:6a:49:a9:c6
>   pubkey:    RSA 2048 bits
>   keyid:     d6:46:be:f5:b4:44:ee:8e:77:64:00:bd:aa:3d:93:6f:53:4f:9f:90
>   subjkey:   5a:42:1a:d4:e3:35:9a:c6:e2:4f:94:82:8b:7e:22:08:6a:49:a9:c6

**Reply by tobiasbrunner** [reply]
The problem is that the R3 intermediate CA certificate you've loaded is not the correct one. As you can see in the output above, the R3 certificate you have is issued by `O=Digital Signature Trust Co., CN=DST Root CA X3` (i.e. [it's the cross-signed one](https://crt.sh/?id=3479778542) that expires in 48 days) and not [the newer one](https://crt.sh/?id=3334561879) issued by `C=US, O=Internet Security Research Group, CN=ISRG Root X1`.

Because the latter root CA certificate is loaded on both client and server and the server receives a certificate request for it, it acts as anchor when building the trust chain. However, no such chain can be built from the server certificate (it stops at the wrong R2 certificate). So the incomplete chain is ignored and only the server certificate is sent to the client.

If you didn't have the root CA certificate installed on the server, the intermediate CA certificate would actually have been sent as there would not have been any known trust anchor and the incomplete trust chain would get accepted. But even if that intermediate CA certificate was sent to the client, and the client had the matching root CA installed, it would only work until it expires in 48 days.

So make sure you load the correct R3 intermediate CA certificate on the server. You could just load Let's Encrypt's `chain.pem`, from which strongSwan will read the first certificate, which is the R3 intermediate (as mentioned, you don't necessarily have to load the root CA certificate on the server, as no complete trust chain is necessary if the server does not recognize any of the CAs received in certificate requests).
